### ISSUE TITLE
Error in webpack "promise" module can not be found

### ISSUE THREAD

**Issue by jhamlet** [graphistry/falcor#2]
@trxcllnt getting the following error when building with webpack:

```
ERROR in ./~/@graphistry/falcor/dist/falcor.min.js
Module not found: Error: Cannot resolve module 'promise' in node_modules/@graphistry/falcor/dist
 @ ./~/@graphistry/falcor/dist/falcor.min.js 4:288-306
```

I can work around it by installing the promise module in my `devDependencies`, but you probably have a dangling `require` somewhere that is making `webpack` try to load it just in case.

**Comment by trxcllnt** [comment]
@jhamlet looks like I forgot to move `promise` from `devDependencies` to `dependencies`. Published `@graphistry/falcor@2.0.3` with a fix.

**Comment by jhamlet** [comment]
@trxcllnt Hmmm... [Netflix/falcor](https://github.com/Netflix/falcor/blob/master/package.json#L71) has the same thing: `promise` as a `devDependency` instead of a regular `dependency`.

I didn't run into this before switching to `@graphistry/falcor` -- just wondering if it's meant to be a dev dependency and the offending code is supposed to be cleaned out when a dist build is created...

**Comment by trxcllnt** [comment]
@jhamlet I think it's a devDependency in netflix/falcor because they've configured browserify to include the `promise` library in the bundle [at compile time](https://github.com/Netflix/falcor/blob/master/build/gulp-build.js#L22). The webpack config for this project [doesn't](https://github.com/graphistry/falcor/blob/master/packages/falcor/build/webpack.config.js#L61) compile the `promise` library into the bundle, but does calls `require("promise")` if the JS environment doesn't have a native `Promise` implementation.

I'll do some investigation into whether I can configure webpack to shim in a Promise implementation for environments without one, hopefully without adding to the bundle size or precluding someone from configuring their build to use a different Promise implementation.

**Comment by jhamlet** [comment]
@trxcllnt I think both browserify and webpack allow for aliasing modules `{ "promise": "some/path/to/my/promise.js" }` so I wouldn't worry too much about it.

Just was worried about the condition I mentioned above.

**Comment by trxcllnt** [comment]
@jhamlet browserify and webpack do allow module aliases, but I don't think they have enough information to alias modules compiled into library "binaries" (since they shorten the names to int keys). Turns out [I can use `es6-promise`](https://gist.github.com/Couto/b29676dd1ab8714a818f#gistcomment-1584602) to shim Promises with just ~7kb extra (minified), so I've published `@graphistry/falcor@2.0.5` with the shim in place.

**Comment by jhamlet** [comment]
Fixed with the latest update.
